What's new in DevTools (112) (#5583)
GoogleChrome/developer.chrome.com: site/en/blog/new-in-devtools-112/index.md

---
layout: 'layouts/blog-post.njk'
title: "What's New in DevTools (Chrome 112)"
authors:
  - sofiayem
date: 2023-03-08
description: ""
hero: 'image/NJdAV9UgKuN8AhoaPBquL7giZQo1/G5pxOHr2wECkBu8J2ujC.jpg'
alt: 'Recorder replay customization, CSS documentation, ignore-listing irrelevant scripts, enhanced logpoint debugging, and more.'
tags:
  - new-in-devtools
  - devtools
  - chrome-112
---

> <!-- image/NJdAV9UgKuN8AhoaPBquL7giZQo1/G5pxOHr2wECkBu8J2ujC.jpg  -->

Reviewer: ```suggestion
<!--image/dPDCek3EhZgLQPGtEG3y0fTn4v82/gctGASDKBFTUtOQqVq2H.png  -->
```
Generate the stable release banner with Figma. We replace the header image with this when stable release roll out. See similar PR: https://github.com/GoogleChrome/developer.chrome.com/pull/5577

1. Install it, [create a new recording](/docs/devtools/recorder/#record), expand the {% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/gjfZMeLnwzpRfOMfXEMY.svg", alt="Play.", width="20", height="20" %} **Replay** options, and select the extension.
1. Provide an arbitrary name and password (`leeloo` and `multipass` will do). 

{% Video src="video/NJdAV9UgKuN8AhoaPBquL7giZQo1/kfAzPxEF08zwvYYKzP4e.mp4", autoplay="true", muted="true", loop="true", controls="true", class="screenshot" %}

To customize the **Recorder** to your needs and integrate it with your tools, consider developing your own extension: explore the [chrome.devtools.recorder API](/docs/extensions/reference/devtools_recorder/) and check out more [extension examples](https://github.com/puppeteer/replay/tree/main/examples/).

{# https://chrome-internal.googlesource.com/devtools/devtools-internal/+/c2102177581f1c74d38502f469d99b20c1835b1c #}
{# https://chrome-internal.googlesource.com/devtools/devtools-internal/+/e304e064dbead1d684b5c61f4fb308b101b4a66b #}

Chromium issue: [1400243](https://crbug.com/1400243).

### Record with pierce selectors {: #pierce-selectors }

> In addition to custom, CSS, ARIA, text, and XPath selectors, you can now record using [pierce selectors](https://pptr.dev/guides/query-selectors#pierce-selectors-pierce). These selectors behave like CSS ones but can also pierce through shadow roots.

Author: Got it.

The tooltip also has a **Learn more** link that takes you to an [MDN CSS Reference](https://developer.mozilla.org/docs/Web/CSS/Reference) on this property.

If you know CSS well, you might find the tooltips bothersome. To turn them all off, check {% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/hmp8j3HiLMCcqPArD9yt.svg", alt="Checkbox.", width="22", height="22" %} **Don't show**.

To turn them back on, check {% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/9gzXiTYY0nZzBxGI6KrV.svg", alt="Settings.", width="24", height="24" %} **Settings** > **Preferences** > **Elements** > {% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/hmp8j3HiLMCcqPArD9yt.svg", alt="Checkbox.", width="22", height="22" %} **Show CSS documentation tooltip**.

{% Aside %}
DevTools pulls the descriptions for tooltips from [VS Code Custom Data](https://github.com/microsoft/vscode-custom-data).
{% endAside %}

{# https://chromium.googlesource.com/devtools/devtools-frontend/+/f5266ee227449dbbc3bc599df1b38cdb36cae4cb #}
{# https://chromium.googlesource.com/devtools/devtools-frontend/+/d4748c98971bfff697f209fe11de892a5b93aca6 #}

Chromium issue: [1401107](https://crbug.com/1401107).

### CSS nesting support {: #nesting }

> The **Elements** > **Styles** pane now recognizes [CSS Nesting](https://w3c.github.io/csswg-drafts/css-nesting/) syntax and applies nested styles to the right elements.

Reviewer: ```suggestion
The **Elements** > **Styles** pane now recognizes [CSS Nesting](/articles/css-nesting/) syntax and applies nested styles to the right elements.
```

Click the link next to the breakpoint message to jump directly to the breakpoint editor.

{% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/8lAz0lb168HXKvhscP2Q.png", alt="The anchor link next to a logpoint message that opens the breakpoint editor.", width="800", height="811" %}

{# https://chromium.googlesource.com/devtools/devtools-frontend/+/c845a441b0fe05c22f88cdb23463edee2b5985b7 #}
{# https://chromium.googlesource.com/devtools/devtools-frontend/+/9762db476cd7414d3ce351f32a0564421f66901f #}
{# https://chromium.googlesource.com/devtools/devtools-frontend/+/42448cc63567ac407fd2088597da83aff17c5b55 #}
{# https://chromium.googlesource.com/devtools/devtools-frontend/+/4739f48e50d41025aba3c2af94e61cc3069aa563 #}
{# https://chromium.googlesource.com/devtools/devtools-frontend/+/bb0e41ed3c30bd988c49a76f0cf084f58c0bddc2 #}

Chromium issue: [1027458](https://crbug.com/1027458).

## Ignore irrelevant scripts during debugging {: #ignore-list }

To help you focus on the most important parts of your code, you can now add irrelevant scripts to the **Ignore List** right from the file tree on the **Sources** > **Page** pane. The debugger ignores all scripts added to the list. 

> Right-click any script or folder and select one of the ignore-related options.

Reviewer: ```suggestion
Right-click any script or folder and select one of the ignore-related options.
```
Explain what happen after we ignore a script. It reduce the frame in stacktrace and call stack, we hide frames that are ignored. We might have a docs already. Link to that. 

Below is the relevant previous WNDT.
https://developer.chrome.com/blog/new-in-devtools-106/#call-stack
https://developer.chrome.com/blog/new-in-devtools-106/#auto-ignore

{% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/RrL7ZmzMjfhtH4gUW3ST.png", alt="Context menus of a folder and script with ignore-related options.", width="800", height="521" %}

Depending on the script or folder you right-clicked, you may see options to add or remove to and from the list:

- A script or a folder to or from a custom exclusion rule.

> - All known third-party scripts.

Reviewer: ```suggestion
```

{# https://chromium.googlesource.com/devtools/devtools-frontend/+/e95d2f3fd27301945a1a095bae4bbcad57326cd8 #}
{# https://chromium.googlesource.com/devtools/devtools-frontend/+/736762eda6a6f30d0e9c383998624e53ee04a6e2 #}
{# https://chromium.googlesource.com/devtools/devtools-frontend/+/2257f7bca42753d744b56f5b99b461a6f0494131 #}

Chromium issue: [883325](https://crbug.com/883325).

## JavaScript Profiler deprecation started {: #js-profiler-deprecation }

As early as [Chrome 58](/blog/devtools-javascript-cpu-profile-migration-2/), the DevTools team planned to eventually deprecate the **JavaScript Profiler** and have Node.js and Deno developers use the **Performance** panel for profiling JavaScript CPU performance.

> 

Reviewer: Need a line to call out for Node.js profiling.

Author: What do you mean? I thought that did call it out.
